content/articles/client-server-react-django-recaptcha/index.md

```diff
@@ -0,0 +1,357 @@
+- Accept the reCAPTCHA's terms of service and click submit.
+- You will then be provided with a **site key**(to use in the frontend) and a **secret key**(to use in the backend).
+
+#### Setting up our Frontend
+Create a folder named **react-django-recaptcha**. Open the folder in your terminal and run the following command:
+
+```bash
+npx create-react-app frontend
+```
+
+The above command will create our React frontend.
+Run the following commands to navigate into the frontend directory and open the directory in your code editor, in this case VS code.
+```bash
+cd frontend
+code .
+```
+
+After you open the project in the code editor, launch the integrated terminal and type in the following command:
+```bash
+npm i react-google-recaptcha
+```
+This will install the Google reCAPTCHA v2 package into our project. We will use this to set up reCAPTCHA in the frontend.
+
+Now, open the `App.js` file and modify the code in it with the following code:
+
+```javascript
+import './App.css';
+import { useState } from 'react';
+import ReCAPTCHA from "react-google-recaptcha";
+
+function App() {
+
+  const [name, setName] = useState('')
+  const [email, setEmail] = useState('')
+  const [message, setMessage] = useState('')
+  const [captchaResult, setCaptchaResult] = useState()
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+
+    // Handle form submission here
+
+  }
+
+     const handleRecaptcha = (value) => {
+
+       fetch('http://127.0.0.1:8000/recaptcha/', {
+         method: 'POST',
+         body: JSON.stringify({ 'captcha_value': value }),
+         headers: { 'Content-Type': 'application/json' }
+       })
+        .then(res => res.json())
+        .then(data => {
+          console.log(data.captcha.success)
+          setCaptchaResult(data.captcha.success)
+        }) 
+     }
+  
+
+
+  return (
+    <div className="container">
+      <h2>Contact us:</h2>
+      <form>
+        <label>Name</label>
+        <input type="text" name="name" placeholder="Your name.." value={name} onChange={(e) => {setName(e.target.value)}} />
+
+        <label>Email</label>
+        <input type="email" name="email" placeholder="example@domain.com" value={email} onChange={(e) => {setEmail(e.target.value)}} />
+
+        <label>Subject</label>
+        <textarea name="subject" placeholder="Write something.." value={message} onChange={(e) => {setMessage(e.target.value)}} />
+
+        <div className="cta">
+          <ReCAPTCHA
+            sitekey="YOUR SITE KEY"
+            onChange={handleRecaptcha}
+          />
+          
+          {
+             captchaResult && <button type='submit' onClick={(e) => {handleSubmit(e)}}>Submit</button>
+          }
+        </div>
+  </form>
+</div>
+  );
+}
+
+export default App;
+```
+
+**Code explanation**
+
+- In the above code, we have a contact form that contains the fields; name, email, and message. We keep track of the data entered in these fields using the useState() hook.
```

Comment:
Have all methods enclosed with  backticks (` `) i.e the `useState()` hook.Update where applicable on the entire article.